File: src/main/java/PlusPart.java
import java.util.ArrayList;
import java.util.Scanner;

public class PlusPart implements Runnable{
    static ArrayList<PlusPart> plusParts = new ArrayList<>();

    private Thread thread;
    private long start;
    private long end;
    private int x;
    private String threadName;
    private long ans;

    PlusPart(long start,long end, int x,String name) {
        this.start = start;
        this.end = end;
        threadName = name;
        this.x = x;
        this.ans = 0;
    }

    public long getAns() {
        return ans;
    }

    @Override
    public void run() {
        for (long i=start;i<end;i++)
            if (contain(i,x)) {
                ans += i;
            }
    }

    public void start() {
        if (thread == null) {
            thread = new Thread(this, threadName);
            thread.start();
        }
    }

    public boolean isAlive(){
        return thread.isAlive();
    }

    private static boolean contain(long num, int x) {
        return String.valueOf(num).contains(String.valueOf(x));
    }

    private static boolean isFinished() {
        for (PlusPart p: plusParts) {
            if (p.isAlive()) return false;
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int x = scanner.nextInt();
        long ans = 0;
        for (long i=0;i<1000000001;i+=100000)
            plusParts.add(new PlusPart(i,i+100000,x,String.format("thread%d",i)));
        for (PlusPart p: plusParts) {
            p.start();
            System.out.println("Thread:"+ p.threadName + " start");
        }
        while(!isFinished());
        for (PlusPart p: plusParts) {
            ans += p.getAns();
        }
        if (x == 0 || x == 1)
            ans -= 1000000000;
        System.out.println(ans);
    }
}

File: src/main/java/Version_Kotlin/PlusPart.kt
import java.util.*
import kotlin.collections.ArrayList
import kotlin.concurrent.thread

class PlusPart (private val start: Long,private val end: Long, private val x: Int):Runnable {
    var ans = 0L;
    var isAlive = false

    override fun run() {
        isAlive = true
        println(start)
        for (j in start until end)
            if (contain(j, x))
                ans += j
        isAlive = false
    }
}

fun main(args: Array<String>) {
    var ans = 0L
    val scanner = Scanner(System.`in`)
    val x: Int = scanner.nextInt()
    val ThreadList = ArrayList<PlusPart>()
    var i = 0L
    while (i < 1000000001) {
        ThreadList.add(PlusPart(i,i+100000,x))
        i += 100000
    }
    for (part in ThreadList)
        part.run()

    var isFinished = false
    while (!isFinished) {
        var finish = true
        for (part in ThreadList) {
            if (part.isAlive) {
                finish = false
                break
            }
        }
        isFinished = finish
    }
    for (p in ThreadList) {
        ans += p.ans
    }
    if (x == 0 || x == 1) ans -= 1000000000
    println(ans)
}

fun contain(num:Long, x:Int) = num.toString().contains(x.toString())